### PR TITLE
update status and implement scale target

### PR DESCRIPTION
**Pull request by qmhu**
1. update ahpa status based on reconciler loop
2. implement scale strategy Manual

### CODE REVIEW

File: pkg/controller/hpa/advanced_hpa_controller.go [hunk 1]
```diff
@@ -43,25 +46,92 @@ func (p *AdvancedHPAController) Reconcile(ctx context.Context, req ctrl.Request)
...
 	}
 
-	// todo: update status
-	// autoscalingStatusOrigin := autoscaler.Status.DeepCopy()
+	setCondition(newStatus, autoscalingapi.Ready, metav1.ConditionTrue, "AdvancedHorizontalPodAutoscalerReady", "Advanced HPA is ready")
+	return ctrl.Result{}, p.UpdateStatus(ctx, ahpa, newStatus)
+}
+
+func (p *AdvancedHPAController) UpdateStatus(ctx context.Context, ahpa *autoscalingapi.AdvancedHorizontalPodAutoscaler, newStatus *autoscalingapi.AdvancedHorizontalPodAutoscalerStatus) error {
+	if !equality.Semantic.DeepEqual(&ahpa.Status, newStatus) {
```
Comment (mfanjie) [inline pkg/controller/hpa/advanced_hpa_controller.go:120]: not a comment but a question, what's the different of this function and reflect.DeepEqual?

File: pkg/controller/hpa/advanced_hpa_controller.go [hunk 2]
```diff
@@ -43,25 +46,92 @@ func (p *AdvancedHPAController) Reconcile(ctx context.Context, req ctrl.Request)
...
+		p.Log.Info("AdvancedHorizontalPodAutoscaler status should be updated", "currentStatus", &ahpa.Status, "newStatus", newStatus)
```
Comment (mfanjie) [inline pkg/controller/hpa/advanced_hpa_controller.go:121]: can we always set log level for info?
and if you are dumping the entire status, the level should be at least 4.

File: pkg/controller/hpa/advanced_hpa_controller.go [hunk 3]
```diff
@@ -43,25 +46,92 @@ func (p *AdvancedHPAController) Reconcile(ctx context.Context, req ctrl.Request)
...
+
+		ahpa.Status = *newStatus
+		err := p.Status().Update(ctx, ahpa)
+		if err != nil {
+			p.Recorder.Event(ahpa, v1.EventTypeNormal, "FailedUpdateStatus", err.Error())
+			p.Log.Error(err, "Failed to update status", "advanced-hpa", klog.KObj(ahpa))
+			return err
+		}
+
+		p.Log.Info("Update AdvancedHorizontalPodAutoscaler status successful", "advanced-hpa", klog.KObj(ahpa))
```
Comment (mfanjie) [inline pkg/controller/hpa/advanced_hpa_controller.go:131]: do we have a short name `ahpa` for the object, if yes, we can simply dump ahpa.

File: pkg/controller/hpa/advanced_hpa_controller.go (hunk 1 above)
Comment (qmhu) [reply pkg/controller/hpa/advanced_hpa_controller.go:120]: reflect.DeepEqual will not necessarily return true for semantically equal quantity objects. That is the purpose of the semantic equality DeepEqual method (apiequality.Semantic.DeepEqual, see https://godoc.org/k8s.io/apimachinery/pkg/api/equality)

File: pkg/controller/hpa/advanced_hpa_controller.go (hunk 3 above)
Comment (qmhu) [reply pkg/controller/hpa/advanced_hpa_controller.go:132]: klog.KObj will format the object like "{namespace}/{name}".

**Pull request merged**